src/caches/rkc_caches.jl

```diff
@@ -88,3 +88,39 @@ end
 function alg_cache(alg::ROCK4,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,uprev2,f,t,dt,reltol,p,calck,::Type{Val{false}})
   ROCK4ConstantCache(real(uEltypeNoUnits), real(uEltypeNoUnits),u)
 end
+
+mutable struct RKCConstantCache{zType} <: OrdinaryDiffEqConstantCache
+  #to match the types to call maxeig!
+  zprev::zType
+  function RKCConstantCache(zprev)
```

Comment:
You don't need to write a constructor explicitly.

Reply:
But every constant cache has one!

Reply:
```julia
julia> struct A{T}
           a::T
       end

julia> A(1)
A{Int64}(1)

julia> struct B{T}
           b::T
           B(x) = new{typeof(x)}(x)
       end

julia> B(1)
B{Int64}(1)
```
You don't need to add a trivial constructor. Julia adds it automatically.